network/network_if_linux.c

```diff
@@ -571,6 +571,105 @@ int32_t fdo_con_disconnect(fdo_con_handle handle)
 	return ret;
 }
 
+/*
+ * Check the REST header for given REST response buffer and offset.
+ *
+ * @param[in] curl_buf: Input buffer that contains the REST header
+ * @param[in] header_start_offset: offset in the buffer that points to the start of REST header
+ * @retval true if header is valid and complete and false otherwise.
+ */
+bool has_header(char *buf,
+		size_t header_start_offset)
+{
+	char tmp[REST_MAX_MSGHDR_SIZE];
+	size_t cur_offset = header_start_offset;
+	bool ret = false;
+	for (;;) {
+		if (memset_s(tmp, sizeof(tmp), 0) != 0) {
```

Comment:
Would the behavior of memset vary based on any platform properties?

Reply:
As far as I am aware, it behaves the same on all platforms.